### PR TITLE
Purge Companies Logo API

### PR DESCRIPTION
**Pull request by abinashpa**
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have added automated tests for my code

### CODE REVIEW

File: app/models/client.rb [hunk 1]
```diff
@@ -131,6 +131,12 @@ def register_on_stripe!
 
   private
 
+    def unique_client_name_for_company
+      if company.clients.where(name:).exists?
```
Comment (akhilgkrishnan) [inline app/models/client.rb:135]: Do we have this requirement?

File: spec/requests/internal_api/v1/companies/purge_logo/destroy_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+RSpec.describe "InternalApi::V1::Companies::PurgeLogo#destroy", type: :request do
+  let(:company) { create(:company, :with_logo) }
+  let(:user) { create(:user, current_workspace: company) }
+
+  before do
+    sign_in user
+    send_request(:delete, "/internal_api/v1/companies/#{company.id}/purge_logo")
+  end
+
+  it "successful status" do
```
Comment (akhilgkrishnan) [inline spec/requests/internal_api/v1/companies/purge_logo/destroy_spec.rb:14]: add failure test

File: app/models/client.rb (hunk 1 above)
Comment (abinashpa) [reply app/models/client.rb:135]: Yes, but in other PR.

**Review by apoorv1316 — APPROVED** [review]
LGTM after adding a failing condition test

File: spec/requests/internal_api/v1/companies/purge_logo/destroy_spec.rb (hunk 2 above)
Comment (abinashpa) [reply spec/requests/internal_api/v1/companies/purge_logo/destroy_spec.rb]: Added for non-admin requests.

**Review by akhilgkrishnan — APPROVED** [review]

**Comment by supriya3105** [comment]
@abinashpa The CI checks are still failing. Please check . Shalaka has fixed the reports related tests that were failing earlier.

**Pull request merged**